pkgs/by-name/wo/wordfence-cli/package.nix

```diff
@@ -0,0 +1,93 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  pkg-config,
+  pcre,
+  hyperscan,
+  testers,
+  runCommand,
+}:
+
+let
+  self = python3Packages.buildPythonApplication rec {
+    pname = "wordfence-cli";
+    version = "5.0.1";
+
+    src = fetchFromGitHub {
+      owner = "wordfence";
+      repo = "wordfence-cli";
+      rev = "v${version}";
+      sha256 = "13hb1xikm8824hhfmv9ggmmkp637yfkf7z58338p66z9l7hsjq1i";
+    };
+
+    nativeBuildInputs = with python3Packages; [
+      pkg-config
+      setuptools
+      pip
+      pcre
+      hyperscan
+    ];
+
+    propagatedBuildInputs = with python3Packages; [
+      packaging
+      requests
+      pymysql
+    ];
+
+    patches = [
+      ./library.patch
+    ];
+
+    postPatch = ''
+      substituteInPlace wordfence/util/library.py \
+        --replace '@PCRE_OUT@' "${pcre.out}" \
+        --replace '@HS_OUT@'  "${hyperscan.out}"
+    '';
+
+    format = "pyproject";
+    doCheck = false;
+
+    passthru.tests = {
+      version = testers.testVersion {
+        package = self;
+        command = "wordfence --version";
+        version = "Wordfence CLI ${version}";
+      };
+
+      pcreSupported =
+        runCommand "wordfence-cli-pcre-supported"
+          {
+            nativeBuildInputs = [ self ];
+          }
+          ''
+            wordfence --version | grep -F "PCRE Supported: Yes"
+            touch $out
+          '';
+
+      vectorscanSupported =
+        runCommand "wordfence-cli-vectorscan-supported"
+          {
+            nativeBuildInputs = [ self ];
+          }
+          ''
+            wordfence --version | grep -F "Vectorscan Supported: Yes"
+            touch $out
+          '';
+    };
+
+    meta = with lib; {
+      description = "Wordfence malware and vulnerability scanner command line utility";
+      homepage = "https://github.com/wordfence/wordfence-cli";
+      changelog = "https://github.com/wordfence/wordfence-cli/releases";
+      license = licenses.gpl3;
+      maintainers = with maintainers; [
+        am-on
+      ];
+      platforms = [
+        "x86_64-linux"
+      ];
```

Comment:
My guess is this isn't actually limited to x86 linux?
```suggestion
      platforms = lib.platforms.linux;
```

Reply:
The limiting factor is platforms of `hyperscan`. 

- `pcre` platforms:
  - aarch64-darwin
  - aarch64-linux
  - i686-linux
  - x86_64-darwin
  - x86_64-linux
 - `hyperscan` platforms:
   - x86_64-darwin
   - x86_64-linux 

`pcre` is required for `malware scan` command, `hyperscan` (used as a faster alternative to `pcre`) could be made optional and then all platforms from `pcre` could be supported. 

I wanted to keep it simple for start as I am not experienced with packaging. 

Will try to tackle it when I have time. Thank you for all the tips and help!

Reply:
I see, well just leave it as is then.

Reply:
I filed https://github.com/wordfence/wordfence-cli/issues/324 to see if they can adopt the `vectorscan` fork of hyperscan.

Reply:
Update: they support vectorscan natively, so let's use that.